rdt/__init__.py

```diff
@@ -92,3 +93,81 @@ def get_demo(num_rows=5):
             'dollars_spent': dollars_spent
         })
     ], ignore_index=True)
+
+
+def _find_addons():
+    """Find and load add-ons based on the given group.
+
+    Args:
+        group (str):
+            The name of the entry points group to load.
+        parent_globals (dict):
+            The caller's global scope. Modules will be added to the parent's global scope through
+            their name.
+        add_all (bool):
+            Whether to also add everything in the add-on's ``module.__all__`` to the parent's
+            global scope. Defaults to ``False``.
+    """
+    group = 'rdt_modules'
+    for entry_point in iter_entry_points(group=group):
+        try:
+            addon = entry_point.load()
+        except Exception:  # pylint: disable=broad-exception-caught
+            msg = f'Failed to load "{entry_point.name}" from "{entry_point.module_name}".'
+            warnings.warn(msg)
+            continue
+
+        module_path, _, object_path = entry_point.name.partition(':')
+        module_path = module_path.split('.')
+
+        if module_path[0] != __name__:
+            msg = (
+                f"Failed to load '{entry_point.name}'. Expected base module to be '{__name__}'"
+                f", found '{module_path[0]}'."
+            )
+            warnings.warn(msg)
+            continue
+
+        base_module = modules[__name__]
+        for depth, submodule in enumerate(module_path[1:-1]):
+            try:
+                base_module = getattr(base_module, submodule)
+            except AttributeError:
+                msg = (
+                    f"Failed to load '{entry_point.name}'. Target submodule "
+                    f"'{'.'.join(module_path[:depth + 2])}' not found."
+                )
+                warnings.warn(msg)
+                continue
+
+        if not hasattr(base_module, module_path[-1]):
+            if object_path:
+                msg = (
+                    f"Failed to load '{entry_point.name}'. Cannot add '{object_path}' to "
+                    f"unknown submodule '{'.'.join(module_path)}'."
+                )
+                warnings.warn(msg)
+                continue
+
+            setattr(base_module, module_path[-1], addon)
+        else:
+            base_module = getattr(base_module, module_path[-1])
+
+        if object_path:
+            split_object = object_path.split('.')
+            try:
+                base_object = base_module
+                if len(split_object) > 1:
+                    base_object = attrgetter('.'.join(split_object[:-1]))(base_module)
+
+                setattr(base_object, object_path[-1], addon)
+            except AttributeError:
+                msg = (
+                    f"Failed to load '{entry_point.name}'. Cannot find "
+                    f"'{'.'.join(split_object[:-1])}' in submodule '{'.'.join(module_path)}'."
+                )
+                warnings.warn(msg)
+                continue
```

Comment:
I would suggest to move all this to a `process_entry_point` function. And maybe try to even split that into multiple steps if possible.